pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -586,6 +602,14 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 					}
 				}
 
+				if noneMode &&
+					service.Resolution == model.ClientSideLB { // TODO: more conditions
```

Comment:
this applies to DNSLB as well.

Reply:
Oh, you mean user declares DNSLB, and we generate config to use DNS cluster ?
It may work. We can add this in separate PR, no need to do all modes in a single PR.